tensorzero-core/src/providers/helpers.rs

```diff
@@ -344,6 +359,109 @@ fn parse_index(s: &str) -> Option<usize> {
+                        // since `value`` is already an object where the target pointer doesn't exist
+                        return Ok(());
+                    }
+                },
+                Value::Array(list) => {
+                    match parse_index(&token) {
+                        Some(index) => {
+                            if let Some(target) = list.get_mut(index) {
+                                value = target;
+                            } else {
+                                tracing::warn!("Skipping deletion of extra_body pointer `{pointer}` - index `{token}` out of bounds");
+                                // If a parent of our target pointer doesn't exist, then do nothing,
+                                // since `value`` is already an object where the target pointer doesn't exist
+                                return Ok(());
+                            }
+                        }
+                        None => {
+                            tracing::warn!("Skipping deletion of extra_body pointer `{pointer}` - non-numeric array index `{token}`");
+                            // If a parent of our target pointer doesn't exist, then do nothing,
+                            // since `value`` is already an object where the target pointer doesn't exist
+                            return Ok(());
+                        }
+                    }
+                }
+                other => {
+                    return Err(Error::new(ErrorDetails::ExtraBodyReplacement {
+                        message: format!(
+                            "Can only index into object or array - found target {other}"
+                        ),
+                        pointer: pointer.to_string(),
+                    }))
+                }
+            }
+        } else {
+            match value {
+                Value::Object(map) => {
+                    if map.remove(&token).is_none() {
+                        tracing::warn!("Skipping deletion of extra_body pointer `{pointer}` - key `{token}` doesn't exist");
+                    }
+                }
+                Value::Array(list) => match parse_index(&token) {
+                    Some(index) => {
+                        if index < list.len() {
+                            list.remove(index);
+                        } else {
+                            tracing::warn!("Skipping deletion of extra_body pointer `{pointer}` - index `{token}` out of bounds");
+                        }
+                    }
+                    None => {
+                        tracing::warn!("Skipping deletion of extra_body pointer `{pointer}` - non-numeric array index `{token}`");
+                    }
+                },
+                other => {
+                    return Err(Error::new(ErrorDetails::ExtraBodyReplacement {
+                        message: format!(
+                            "Can only index into object or array - found target {other}"
```

Comment:
shouldn't we also just log a warning and continue if it's a value? that seems like the thing being pointed at is missing in that case as well no?